### PR TITLE
Spawner wouldn't spawn Types with optional parameters in ctor...

### PR DESCRIPTION
**Pull request by Reetus**
Spawner wouldn't spawn Types with optional parameters in ctor due to parameter length check

### CODE REVIEW

File: Projects/Scripts/Engines/Spawner/Spawner.cs [hunk 1]
```diff
@@ -472,49 +473,59 @@ public bool Spawn(SpawnerEntry entry, out EntryFlags flags)
...
+                  {
+                    o = ctor.Invoke(paramValues);
+                    break;
+                  }
                 }
               }
             }
           }
 
+          for (int j = 0; j < realProps.Length; j++)
```
Comment (kamronbatman) [inline Projects/Scripts/Engines/Spawner/Spawner.cs]: Not nested so it can be j?

File: Projects/Scripts/Engines/Spawner/Spawner.cs [hunk 2]
```diff
@@ -472,49 +473,59 @@ public bool Spawn(SpawnerEntry entry, out EntryFlags flags)
...
+          {
+            if (realProps[ j ] != null)
```
Comment (kamronbatman) [inline Projects/Scripts/Engines/Spawner/Spawner.cs]: While we don't have formal linting, I try to stick to no spaces between index accessors.

File: Projects/Server/Utilities/ActivatorUtil.cs [hunk 3]
```diff
@@ -42,32 +54,44 @@ public static ConstructorInfo GetConstructor(Type type, params Type[] args)
...
       catch (Exception e)
       {
         throw new TypeInitializationException(type.ToString(), e);
       }
     }
-    public static object CreateInstance(Type type)
+
+    private static ConstructorInfo CheckPredicate(ConstructorInfo ci, Predicate<ConstructorInfo> predicate)
+    {
+      if (predicate == null || predicate.Invoke(ci))
```
Comment (kamronbatman) [inline Projects/Server/Utilities/ActivatorUtil.cs]: Can be compressed to `predicate?. Invoke(ci) != false`

File: Projects/Server/Utilities/ActivatorUtil.cs [hunk 4]
```diff
@@ -42,32 +54,39 @@ public static ConstructorInfo GetConstructor(Type type, params Type[] args)
...
       }
       catch (Exception e)
       {
         throw new TypeInitializationException(type.ToString(), e);
       }
     }
-    public static object CreateInstance(Type type)
+
+    private static ConstructorInfo CheckPredicate(ConstructorInfo ci, Predicate<ConstructorInfo> predicate)
+    {
```
Comment (flamewave000) [inline Projects/Server/Utilities/ActivatorUtil.cs]: This function seems unnecessary to me. When ever you want to check the predicate just do the following `predicate?.Invoke(ci) == true` or for a fail check: `predicate?.Invoke(ci) != true`
The only place I see the returned object used is on line 34. But that should probably be broken down into a proper if-else statement for clarity.

**Comment by Reetus** [comment]
Everyone should be happy

**Pull request merged**